Week3/README.md

```diff
@@ -1,57 +1,82 @@
+CREATE DATABASE IF NOT EXISTS supermarket;
 
-Objective: Students should be able to create an entity relationship diagram based on a qualatative description of data requirements, and translate that into a MySQL database schema. Students should also be able to compare and contrast relational (like MySQL) and NoSQL databases (considering their benefits and drawbacks).
+USE supermarket;
 
-## Pre-Class Readings
+CREATE TABLE IF NOT EXISTS supermarket.products(
+  id INT NOT NULL AUTO_INCREMENT,
+  name VARCHAR(45) NOT NULL
+  unit VARCHAR(15) NOT NULL,
+  minimum_quantity_allowed INT NOT NULL,
+  PRIMARY KEY (id))
+  ENGINE = DB;
 
-Before arriving to class on Sunday, please watch all of the videos in [this video playlist](https://www.lynda.com/SharedPlaylist/ae29ea2f495c432793abc220da47baa6) on Lynda.
-- Choosing Primary Keys
-- Defining One-to-Many Relationships
-- Exploring One-to-One Relationships
-- Exploring Many-to-Many Relationships
-- Understanding Relationship Rules and Referential Integrity
-- Defining Table Relationships
-- NoSQL databases
-- GraphQL: Introduction and History
-- Why use GraphQL?
+CREATE TABLE IF NOT EXISTS supermarket.supplier (
+  id INT NOT NULL AUTO_INCREMENT,
+  name VARCHAR(45) NOT NULL,
+  mobilephone VARCHAR(20) NOT NULL,
+  localphone VARCHAR(20) NULL,
+  email VARCHAR(45) NOT NULL,
+  address VARCHAR(45) NOT NULL,
+  PRIMARY KEY (id))
+  ENGINE = DB;
 
-Also, please read the following page that explains database foreign keys.
-- [What is a Database Foreign Key](http://databases.about.com/cs/specificproducts/g/foreignkey.htm)
+CREATE TABLE IF NOT EXISTS supermarket.purchase (
+  id INT NOT NULL AUTO_INCREMENT,
+  date DATETIME NOT NULL,
+  supplier_id INT NOT NULL,
+  PRIMARY KEY (id, supplier_id),
+  INDEX fk_purchase_supplier_idx (supplier_id ASC),
+  CONSTRAINT fk_purchase_supplier
+  FOREIGN KEY (supplier_id)
+  REFERENCES supermarket.supplier(id)
+  ON DELETE NO ACTION
+  ON UPDATE NO ACTION)
+  ENGINE = DB;
 
-## Main Topics
+CREATE TABLE IF NOT EXISTS supermarket.purchase_products (
+  id INT NOT NULL AUTO_INCREMENT,
+  products_id INT NOT NULL,
+  purchase_id INT NOT NULL,
+  quantity INT NOT NULL,
+  expiration_date DATE NOT NULL,
+  purchasing_price DECIMAL NOT NULL,
+  selling_price DECIMAL NOT NULL,
+  PRIMARY KEY (id , products_id , purchase_id ),
+  INDEX fk_supermarket_products_idx (products_id ASC),
+  INDEX fk_supermarket_purchase 1_idx (purchase_id  ASC),
+  CONSTRAINT fk_supermarket_product1
+  FOREIGN KEY (products_id)
+  REFERENCES supermarket.products (id)
+  ON DELETE NO ACTION
+  ON UPDATE NO ACTION,
+  CONSTRAINT  fk_supermarket_purchase1
+  FOREIGN KEY (purchase_id)
+  REFERENCES supermarket.purchase (id)
+  ON DELETE NO ACTION
```

Comment:
We didn't talk about cascades in class - are you certain you know what this does? I think it's probably correct for this table, but its not necessary unless another table is referencing this table, which is probably not the case - this table is referencing others.